pkg/services/sqlstore/stats.go

```diff
@@ -68,6 +68,9 @@ func GetSystemStats(query *m.GetSystemStatsQuery) error {
 	}
 
 	query.Result = &stats
+
+	bus.Publish(&stats)
```

Comment:
not sure I understand the publish / even handler addition. What was wrong with using the result property?

Reply:
maybe just calling updateTotalStats after metrics init will fix the initial 0 values and make sure the stats have value before the publisher is started.

Reply:
The reason for doing it this way is so that the prometheus stats can be updated every time we pull the stats from the DB, this means that they're automatically updated at startup by the call we are already making in `EnsureAdminUser` and we don't need to immediately call it again.

More generally this is an implementation of the https://en.wikipedia.org/wiki/Observer_pattern so that we split the responsibility for periodically getting the stats out of the DB from the responsibility for updating the in-memory values, and so that the in-memory values are always up to date regardless of what is triggering the stats to be gathered from the DB.

Reply:
Calling `updateTotalStats()` on  https://github.com/grafana/grafana/blob/master/pkg/metrics/metrics.go#L317 seems like a simpler solution. Since the first thing we do after `EnsureAdminUser` is to set up metrics it shouldn't matter execution time wise. 

I think the Observer pattern is nice when you have multiple sources you want to keep updated. In this case we only have two.